reconcile/openshift_tekton_resources.py

```diff
@@ -0,0 +1,373 @@
+import sys
+import logging
+import json
+from typing import Any, Optional, Union
+
+import yaml
+import jinja2
+
+from reconcile import queries
+from reconcile import openshift_base as ob
+from reconcile.status import ExitCodes
+from reconcile.utils import gql
+from reconcile.utils.defer import defer
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+from reconcile.utils.semver_helper import make_semver
+
+LOG = logging.getLogger(__name__)
+QONTRACT_INTEGRATION = 'openshift-tekton-resources'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+# it must be a single character due to resource max length
+OBJECTS_PREFIX = 'o'
+RESOURCE_MAX_LENGTH = 63
+
+# Defaults
+DEFAULT_DEPLOY_RESOURCES_STEP_NAME = 'qontract-reconcile'
+DEFAULT_DEPLOY_RESOURCES = {'requests': {'cpu': '50m',
+                                         'memory': '200Mi'},
+                            'limits': {'cpu': '200m',
+                                       'memory': '300Mi'}}
+# Queries
+SAAS_FILES_QUERY = '''
+{
+  saas_files: saas_files_v2 {
+    path
+    name
+    configurableResources
+    pipelinesProvider {
+      name
+      provider
+    }
+    deployResources {
+      requests {
+        cpu
+        memory
+      }
+      limits {
+        cpu
+        memory
+      }
+    }
+  }
+}
+'''
+
+
+class OpenshiftTektonResourcesNameTooLongError(Exception):
+    pass
+
+
+class OpenshiftTektonResourcesBadConfigError(Exception):
+    pass
+
+
+def fetch_saas_files(saas_file_name: Optional[str]) -> list[dict[str, Any]]:
+    '''Fetch saas files that can be handled by this integration: those with
+    configurableResources set to True'''
+    saas_files = [
+        s for s in gql.get_api().query(SAAS_FILES_QUERY)['saas_files']
+        if s.get('configurableResources')]
+
+    if saas_file_name:
+        saas_file = None
+        for sf in saas_files:
+            if sf['name'] == saas_file_name:
+                saas_file = sf
+                break
+
+        return [saas_file] if saas_file else []
+
+    return saas_files
+
+
+def fetch_tkn_providers(saas_file_name: Optional[str]) -> dict[str, Any]:
+    '''Fetch tekton providers data for the saas files handled here'''
+    saas_files = fetch_saas_files(saas_file_name)
+    if not saas_files:
+        return {}
+
+    duplicates: set[str] = set()
+    all_tkn_providers = {}
+    for pipeline_provider in queries.get_pipelines_providers():
+        if pipeline_provider['provider'] != 'tekton':
```

Comment:
i know this is a copy-paste, but there is a Providers enum in saasherder that can be used

Reply:
I'm not sure I understand what you mean. It's true that the only provider is currently tekton, but it's fine to avoid touching this if we ever have more...

Reply:
https://github.com/app-sre/qontract-reconcile/blob/ff6189e0523e545176fe00ffc343921f2a558d3c/reconcile/utils/saasherder.py#L26

Reply:
ahhh, ok... I was clearly not understanding this 😂 

Fixed in 84892ea